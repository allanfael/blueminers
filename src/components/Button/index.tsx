import React from 'react'
import { ActivityIndicator, ViewStyle } from 'react-native'
import { RectButton } from 'react-native-gesture-handler'
import { useTheme } from '@hooks/useTheme'
import { createStyles } from 'responsive-react-native'

import { Typography } from '../Typography'

interface Props {
  title: string
  styles?: ViewStyle
  loading?: boolean
  onPress(): void
}

export const Button = ({ title, styles, onPress, loading }: Props) => {
  const backgroundColor = useTheme('button')

  return (
    <RectButton
      style={[
        styles,
        style.button,
        {
          backgroundColor,
        },
      ]}
      onPress={onPress}
      enabled={!loading}
    >
      {loading && <ActivityIndicator size="small" color="white" />}
      <Typography variant="normalMedium" color="textButton">
        {title}
      </Typography>
    </RectButton>
  )
}

const style = createStyles({
  button: {
    flexDirection: 'row',
    gap: 8,
    width: '100%',
    height: 56,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    elevation: 3,
  },
})
