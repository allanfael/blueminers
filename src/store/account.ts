import { create } from 'zustand'

export type Account = {
  id: string | null
  user: string | null
  currentBalance: number | null
  accumulatedGain: number | null
  available: number | null
  brokerGain: number | null
  lastIncome: number | null
  lastUpdate: Date | null
  incomePercent: number | null
}

interface AccountState {
  account: Account
  save: (account: Account) => void
  update: (account: Account) => void
  clean: () => void
}

const initialState: Account = {
  user: null,
  id: null,
  currentBalance: null,
  accumulatedGain: null,
  available: null,
  brokerGain: null,
  lastIncome: null,
  incomePercent: null,
  lastUpdate: null,
}

export const useAccountStore = create<AccountState>()((set) => ({
  account: initialState,
  save(account) {
    set(() => ({
      account,
    }))
  },
  update(account) {
    set((state) => ({
      ...state,
      ...account,
    }))
  },
  clean() {
    set(() => ({
      account: initialState,
    }))
  },
}))
