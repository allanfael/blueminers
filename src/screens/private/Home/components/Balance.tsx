import React from 'react'
import { Pressable, View } from 'react-native'
import { RectButton } from 'react-native-gesture-handler'
import { Typography } from '@components/Typography'
import { MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons'
import { useTheme } from '@hooks/useTheme'
import { useAccountStore } from '@store/account'
import { userStore } from '@store/user'
import { createStyles } from 'responsive-react-native'
import { currencyParse } from 'utils/currencyParse'
import { hoursParse } from 'utils/hourParse'

interface Props {
  clipboard(): void
  depositNavigation(): void
}

export const Balance = ({ clipboard, depositNavigation }: Props) => {
  const { account } = useAccountStore()
  const { onShowBalance, showBalance } = userStore()
  const backgroundColor = useTheme('button')

  return (
    <View>
      <Typography variant="smallMedium" color="text">
        Saldo
      </Typography>

      <View style={styles.balance}>
        <Typography variant="LargeBold" style={styles.money} color="text">
          {showBalance ? currencyParse(account.currentBalance) : '****'}
        </Typography>

        <MaterialIcons
          name={showBalance ? 'visibility' : 'visibility-off'}
          size={26}
          color="#666"
          onPress={() => onShowBalance(!showBalance)}
        />
      </View>

      <RectButton
        style={[
          styles.button,
          {
            backgroundColor,
          },
        ]}
        onPress={depositNavigation}
      >
        <Typography variant="smallMedium" color="textButton">
          Depositar
        </Typography>
      </RectButton>

      <Typography variant="smallMedium" color="info" style={styles.incomeGain}>
        Nosso rendimento com mineração foi de{' '}
        {account.incomePercent?.toString()}%
      </Typography>

      <Typography
        variant="smallMedium"
        color="info"
        style={styles.updatedMessage}
      >
        Ultima atualização {hoursParse(account.lastUpdate)}
      </Typography>

      <Pressable style={styles.inviteCard} onPress={clipboard}>
        <Typography variant="normalMedium" color="white">
          Link de indicação
        </Typography>
        <MaterialCommunityIcons name="content-copy" size={16} color="#fff" />
      </Pressable>
    </View>
  )
}

const styles = createStyles({
  title: { marginTop: 40 },
  balance: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  money: {
    marginTop: 6,
    marginRight: 30,
  },
  button: {
    height: 44,
    width: 140,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 22,
    marginTop: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    elevation: 3,
  },
  inviteCard: {
    padding: 16,
    alignItems: 'center',
    alignContent: 'center',
    borderRadius: 12,
    backgroundColor: '#0c7c65',
    marginTop: 30,
    marginBottom: 30,
    flexDirection: 'row',
    gap: 10,
  },
  lastIncome: {
    marginLeft: 12,
  },
  incomeGain: {
    marginTop: 40,
  },
  updatedMessage: {
    marginTop: 16,
    fontStyle: 'italic',
  },
})
