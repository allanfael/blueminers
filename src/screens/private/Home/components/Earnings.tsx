import React from 'react'
import { View } from 'react-native'
import { RectButton } from 'react-native-gesture-handler'
import { Typography } from '@components/Typography'
import { useTheme } from '@hooks/useTheme'
import { useAccountStore } from '@store/account'
import { userStore } from '@store/user'
import { createStyles } from 'responsive-react-native'
import { hasMoney } from 'utils/ableToWithdraw'
import { currencyParse, roundedValue } from 'utils/currencyParse'

interface Props {
  withdrawNavigation(): void
}

export const Earnings = ({ withdrawNavigation }: Props) => {
  const { showBalance } = userStore()
  const { account } = useAccountStore()

  const backgroundColor = useTheme('button')

  const hasBlucoin = account.blucoinBalance && account.blucoinBalance > 0

  return (
    <View>
      <Typography variant="LargeBold" color="text" style={styles.title}>
        Meus Ganhos
      </Typography>

      <View style={[styles.card, styles.cardAccGain]}>
        <Typography variant="normalMedium" color="text">
          Ganho acumulado
        </Typography>

        <View style={[styles.row, styles.between]}>
          <Typography variant="mediumBold" style={styles.money} color="text">
            {showBalance ? currencyParse(account.accumulatedGain) : '****'}
          </Typography>
        </View>
      </View>

      <View style={[styles.card, styles.cardLastIncome]}>
        <Typography variant="normalMedium" color="text">
          Rendimento do dia
        </Typography>

        <View style={[styles.row, styles.between]}>
          <Typography variant="mediumBold" style={styles.money} color="text">
            {currencyParse(account.lastIncome)}
          </Typography>
        </View>
      </View>

      <View style={[styles.card, styles.cardAvailable]}>
        <Typography variant="normalMedium" color="text">
          Ganho dispon√≠vel
        </Typography>
        <Typography variant="mediumBold" style={styles.money} color="text">
          {showBalance ? currencyParse(account.available) : '****'}
        </Typography>
        {hasMoney(account.available) && (
          <RectButton
            style={[
              styles.button,
              {
                backgroundColor,
              },
            ]}
            onPress={withdrawNavigation}
          >
            <Typography variant="smallBold" color="textButton">
              Retirar
            </Typography>
          </RectButton>
        )}
      </View>

      {!!hasBlucoin && (
        <View style={[styles.card, styles.cardBlucoin]}>
          <Typography variant="normalMedium" color="text">
            Blucoin
          </Typography>
          <Typography variant="mediumBold" style={styles.money} color="text">
            {roundedValue(account.blucoinBalance)} BLC
          </Typography>
        </View>
      )}

      <View style={[styles.card, styles.cardBroker]}>
        <Typography variant="normalMedium" color="text">
          Ganho ID Broker
        </Typography>
        <Typography variant="mediumBold" style={styles.money} color="text">
          {currencyParse(account.brokerGain)}
        </Typography>
      </View>
    </View>
  )
}

const styles = createStyles({
  title: { marginTop: 40 },
  button: {
    height: 44,
    width: 140,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#1f35d9',
    borderRadius: 22,
    marginTop: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    elevation: 3,
  },
  money: {
    marginTop: 6,
    marginRight: 30,
  },
  cardLastIncome: {
    backgroundColor: '#ff7f5072',
  },
  cardAvailable: {
    backgroundColor: '#de31626c',
  },
  cardAccGain: {
    backgroundColor: '#0d6dfd76',
  },
  cardBlucoin: {
    backgroundColor: '#590dfd67',
  },
  cardBroker: {
    backgroundColor: '#f1c40f68',
    marginBottom: 10,
  },
  card: {
    marginTop: 30,
    width: '100%',
    borderRadius: 14,
    padding: 20,
    justifyContent: 'center',
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
  },
  between: {
    justifyContent: 'space-between',
    alignItems: 'baseline',
  },
})
