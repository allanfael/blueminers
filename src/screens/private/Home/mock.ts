import { Account } from '@store/account'

export const mockHome: Account = {
  user: 'Test',
  id: '123',
  currentBalance: 110,
  accumulatedGain: 110,
  available: 100,
  brokerGain: 10,
  lastIncome: 10,
  incomePercent: 1,
  lastUpdate: new Date('2023-09-30T20:16:58.966Z'),
  blucoinBalance: 10,
  pendingWithdraw: 0,
}
