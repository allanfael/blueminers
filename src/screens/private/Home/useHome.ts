/* eslint-disable react-hooks/exhaustive-deps */
import { useCallback, useEffect, useState } from 'react'
import { useToast } from 'react-native-toast-notifications'
import { api } from '@services/api/api'
import ResponseError from '@services/api/ResponseError'
import { Account, useAccountStore } from '@store/account'
import { userStore } from '@store/user'
import * as Clipboard from 'expo-clipboard'

export type UseHomeProps = {
  error: string
  loading: boolean
  refresh: boolean
  account: Account
  showBalance: boolean
  onShowBalance(value: boolean): void
  clipboard(): void
  update(): void
}

export const useHome = (): UseHomeProps => {
  const { save, account, update: updateStore } = useAccountStore()
  const { logout, onShowBalance, showBalance } = userStore()

  const [error, setError] = useState('')
  const [loading, setLoading] = useState(false)
  const [refresh, setRefresh] = useState(false)

  const toast = useToast()

  const clipboard = async () => {
    if (account.user) {
      const link = `https://bluminers.vercel.app/user/signup?broker=${account.user}`
      await Clipboard.setStringAsync(link)
      toast.show('Link copiado com sucesso')

      return
    }

    toast.show('Não foi possível copiar o link')
  }

  const fetch = useCallback(
    async (
      loading: (loading: boolean) => void,
      saveData: (data: Account) => void,
    ) => {
      try {
        loading(true)
        const home = await api.home()
        const historic = await api.historic()

        const pendingWithdraw = historic.withdrawals.pending.length

        const data = {
          ...home,
          pendingWithdraw,
        }

        saveData(data)
      } catch (e) {
        if (e instanceof ResponseError) {
          return setError(e.message as string)
        }

        logout()
      } finally {
        loading(false)
      }
    },
    [save, logout],
  )

  useEffect(() => {
    fetch(setLoading, save)
  }, [])

  const update = () => fetch(setRefresh, updateStore)

  return {
    error,
    loading,
    refresh,
    clipboard,
    update,
    account,
    showBalance,
    onShowBalance,
  }
}
