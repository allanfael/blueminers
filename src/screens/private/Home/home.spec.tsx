import React from 'react'
import { initialState } from '@store/account'
import { render, screen } from '@testing-library/react-native'

import { Balance } from './components/Balance'
import { mockHome } from './mock'
import { useHome, UseHomeProps } from './useHome'
import { Home } from '.'

jest.mock('./useHome', () => ({
  useHome: jest.fn(),
}))

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  }
})

describe('<Home />', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render Home correctly', async () => {
    ;(useHome as jest.MockedFunction<typeof useHome>).mockReturnValue({
      loading: false,
      error: '',
      refresh: false,
      update: jest.fn(),
      clipboard: jest.fn(),
      account: initialState,
      showBalance: false,
      onShowBalance: jest.fn(),
    } as UseHomeProps)

    render(<Home />)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render Home with values correctly', async () => {
    ;(useHome as jest.MockedFunction<typeof useHome>).mockReturnValue({
      loading: false,
      error: '',
      refresh: false,
      update: jest.fn(),
      clipboard: jest.fn(),
      account: mockHome,
      showBalance: false,
      onShowBalance: jest.fn(),
    } as UseHomeProps)

    render(<Home />)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render Balance component without show balance value', async () => {
    const { getByText } = render(
      <Balance
        account={mockHome}
        showBalance={false}
        onShowBalance={jest.fn()}
        clipboard={jest.fn()}
        depositNavigation={jest.fn()}
      />,
    )

    const currentBalance = await getByText('****')

    expect(currentBalance).toBeDefined()
  })
})
