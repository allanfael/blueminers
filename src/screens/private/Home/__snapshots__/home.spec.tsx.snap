// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render Home correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    [
      undefined,
      undefined,
      {
        "paddingBottom": 0,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 20,
      },
    ]
  }
  refreshControl={
    <RefreshControlMock
      onRefresh={[MockFunction]}
      refreshing={false}
    />
  }
  showsVerticalScrollIndicator={false}
>
  <RCTRefreshControl />
  <View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Saldo
      </Text>
      <View>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 28,
              "lineHeight": 36,
            }
          }
          testID="balanceId"
        >
          ****
        </Text>
        <MaterialIcons
          color="#666"
          name="visibility-off"
          onPress={[Function]}
          size={26}
        />
      </View>
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        delayLongPress={600}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[MockFunction]}
        style={
          {
            "backgroundColor": "#1f35d9",
          }
        }
        underlayColor="black"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "black",
              "borderBottomLeftRadius": undefined,
              "borderBottomRightRadius": undefined,
              "borderRadius": undefined,
              "borderTopLeftRadius": undefined,
              "borderTopRightRadius": undefined,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Muli_700Bold",
              "fontSize": 14,
              "lineHeight": 14,
            }
          }
        >
          Depositar
        </Text>
      </RNGestureHandlerButton>
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Nosso rendimento com mineração foi de
         
        %
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Ultima atualização 
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Link de indicação
        </Text>
        <MaterialCommunityIcons
          color="#fff"
          name="content-copy"
          size={16}
        />
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 28,
            "lineHeight": 36,
          }
        }
      >
        Histórico
      </Text>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              [
                undefined,
                {
                  "backgroundColor": "#e0e0e0",
                },
              ]
            }
          >
            <MaterialIcons
              color="#000"
              name="vertical-align-bottom"
              size={24}
            />
          </View>
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 16,
                "lineHeight": 18,
              }
            }
          >
            Saques
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              [
                undefined,
                {
                  "backgroundColor": "#e0e0e0",
                },
              ]
            }
          >
            <MaterialIcons
              color="#000"
              name="vertical-align-top"
              size={24}
            />
          </View>
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 16,
                "lineHeight": 18,
              }
            }
          >
            Depósitos
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 28,
            "lineHeight": 36,
          }
        }
      >
        Meus Ganhos
      </Text>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho acumulado
        </Text>
        <View
          style={
            [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 22,
                "lineHeight": 28,
              }
            }
          >
            ****
          </Text>
        </View>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Rendimento do dia
        </Text>
        <View
          style={
            [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 22,
                "lineHeight": 28,
              }
            }
          >
            Saldo indisponível
          </Text>
        </View>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho disponível
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 22,
              "lineHeight": 28,
            }
          }
        >
          ****
        </Text>
        <RNGestureHandlerButton
          activeOpacity={0.105}
          collapsable={false}
          delayLongPress={600}
          handlerTag={2}
          handlerType="NativeViewGestureHandler"
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={[MockFunction]}
          style={
            {
              "backgroundColor": "#1f35d9",
            }
          }
          underlayColor="black"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "black",
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": undefined,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <Text
            style={
              {
                "color": "#fff",
                "fontFamily": "Muli_700Bold",
                "fontSize": 14,
                "lineHeight": 14,
              }
            }
          >
            Retirar
          </Text>
        </RNGestureHandlerButton>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho ID Broker
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 22,
              "lineHeight": 28,
            }
          }
        >
          Saldo indisponível
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<Home /> should render Home with values correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    [
      undefined,
      undefined,
      {
        "paddingBottom": 0,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 20,
      },
    ]
  }
  refreshControl={
    <RefreshControlMock
      onRefresh={[MockFunction]}
      refreshing={false}
    />
  }
  showsVerticalScrollIndicator={false}
>
  <RCTRefreshControl />
  <View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Saldo
      </Text>
      <View>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 28,
              "lineHeight": 36,
            }
          }
          testID="balanceId"
        >
          ****
        </Text>
        <MaterialIcons
          color="#666"
          name="visibility-off"
          onPress={[Function]}
          size={26}
        />
      </View>
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        delayLongPress={600}
        handlerTag={3}
        handlerType="NativeViewGestureHandler"
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        onPress={[MockFunction]}
        style={
          {
            "backgroundColor": "#1f35d9",
          }
        }
        underlayColor="black"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "black",
              "borderBottomLeftRadius": undefined,
              "borderBottomRightRadius": undefined,
              "borderRadius": undefined,
              "borderTopLeftRadius": undefined,
              "borderTopRightRadius": undefined,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Muli_700Bold",
              "fontSize": 14,
              "lineHeight": 14,
            }
          }
        >
          Depositar
        </Text>
      </RNGestureHandlerButton>
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Nosso rendimento com mineração foi de
         
        1
        %
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Ultima atualização 
        30/09/2023  17:16:58
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Link de indicação
        </Text>
        <MaterialCommunityIcons
          color="#fff"
          name="content-copy"
          size={16}
        />
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 28,
            "lineHeight": 36,
          }
        }
      >
        Histórico
      </Text>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              [
                undefined,
                {
                  "backgroundColor": "#e0e0e0",
                },
              ]
            }
          >
            <MaterialIcons
              color="#000"
              name="vertical-align-bottom"
              size={24}
            />
          </View>
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 16,
                "lineHeight": 18,
              }
            }
          >
            Saques
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              [
                undefined,
                {
                  "backgroundColor": "#e0e0e0",
                },
              ]
            }
          >
            <MaterialIcons
              color="#000"
              name="vertical-align-top"
              size={24}
            />
          </View>
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 16,
                "lineHeight": 18,
              }
            }
          >
            Depósitos
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "Muli_700Bold",
            "fontSize": 28,
            "lineHeight": 36,
          }
        }
      >
        Meus Ganhos
      </Text>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho acumulado
        </Text>
        <View
          style={
            [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 22,
                "lineHeight": 28,
              }
            }
          >
            ****
          </Text>
        </View>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Rendimento do dia
        </Text>
        <View
          style={
            [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Muli_700Bold",
                "fontSize": 22,
                "lineHeight": 28,
              }
            }
          >
            R$ 10,00
          </Text>
        </View>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho disponível
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 22,
              "lineHeight": 28,
            }
          }
        >
          ****
        </Text>
        <RNGestureHandlerButton
          activeOpacity={0.105}
          collapsable={false}
          delayLongPress={600}
          handlerTag={4}
          handlerType="NativeViewGestureHandler"
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={[MockFunction]}
          style={
            {
              "backgroundColor": "#1f35d9",
            }
          }
          underlayColor="black"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "black",
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": undefined,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <Text
            style={
              {
                "color": "#fff",
                "fontFamily": "Muli_700Bold",
                "fontSize": 14,
                "lineHeight": 14,
              }
            }
          >
            Retirar
          </Text>
        </RNGestureHandlerButton>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Blucoin
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 22,
              "lineHeight": 28,
            }
          }
        >
          10
           BLC
        </Text>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_600SemiBold",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          Ganho ID Broker
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Muli_700Bold",
              "fontSize": 22,
              "lineHeight": 28,
            }
          }
        >
          R$ 10,00
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
